feature-libs/product-configurator/rulebased/cpq/cpq-configurator-overview-normalizer.ts

```diff
@@ -0,0 +1,79 @@
+import { Injectable } from '@angular/core';
+import { Converter } from '@spartacus/core';
+import { Configurator } from '../core/model/configurator.model';
+import { Cpq } from './cpq.models';
+
+@Injectable()
+export class CpqConfiguratorOverviewNormalizer
+  implements Converter<Cpq.Configuration, Configurator.Overview> {
+  constructor() {}
+
+  convert(
+    source: Cpq.Configuration,
+    target?: Configurator.Overview
+  ): Configurator.Overview {
+    const resultTarget: Configurator.Overview = {
+      ...target,
+      productCode: source.productSystemId,
+      groups: source.tabs?.flatMap((tab) => this.convertTab(tab)),
+      totalNumberOfIssues: this.calculateTotalNumberOfIssues(source),
+    };
+    return resultTarget;
+  }
+
+  convertTab(tab: Cpq.Tab): Configurator.GroupOverview {
```

Comment:
Better make protected and access in test like  `expect(serviceUnderTest['convertTab'](tab).attributes.length).toBe(2);`

Reply:
that was new to me (how to access protected methods in tests)